src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -335,10 +337,21 @@ public bool CancellationPending
             {
                 _backgroundWorker.ReportProgress(-1); // Call with an out-of-range percentage will update the value of
                                                       // _cancellationPending but do nothing else.
-                return _cancellationPending;
+
+                return _cancellationPending || (_cancellationSource?.IsCancellationRequested ?? false);
             }
         }
 
+        /// <summary>
+        /// Defines a cancellation token which can be used to stop the task from an external source.
+        /// </summary>
+        [Category("Behavior"), Description("Defines a cancellation token which can be used to stop the task from an external source.")]
+        public CancellationToken CancellationToken
```

Comment:
Let's remove this. The external cancellation token should come via `ShowDialog` or `Show` (2 new overloads).

```
progressDialog.ShowDialog(optionalCancellationTokenProvidedByTheUser);
```